build_examples.sh

```diff
@@ -1,133 +1,42 @@
 #!/usr/bin/env bash
 
-# This script is intended to be run from an Ubuntu 18.04 Docker image.
-# Since it uses `apt-get` to install dependencies required by the examples,
-# it may also run on other versions of Ubuntu, or other Debian-derived distros.
-# It will definitely not run without `apt-get` installed!
+# This script is intended to be run from an Ubuntu Docker image (or other Debian-derived distros).
+# It uses `apt-get` to install dependencies required by the examples,
 
 set -euo pipefail
 
-DEFAULT_BUILD_DIR="build"
-BUILD_DIR="${DEFAULT_BUILD_DIR}"
-SCRIPT_NAME=$(basename $0)
-PROJECT_DIRNAME=$(dirname $(readlink -f $0))
-USAGE="Usage: ${SCRIPT_NAME} [options]...
-
-Options:
-  -h, --help                  Print this message
-  -b, --build-dir  BUILD_DIR  Build examples in BUILD_DIR instead of default '${DEFAULT_BUILD_DIR}'
-  --all                       Build all the examples available
-  --fltk                      Build fltk example
-  --gtkmm                     Build gtkmm example
-  --imgui                     Build imgui example
-  --nana                      Build nana example
-  --qt                        Build qt example
-  --sdl                       Build sdl example
-"
-
-function build_fltk() {
+function install_fltk() {
   apt-get install -y --no-install-recommends libfltk1.3-dev libgl1-mesa-dev fluid
-  BUILD_DIR="${1:-build-fltk}"
-  mkdir -p "${BUILD_DIR}"
-  cmake -S "${PROJECT_DIRNAME}" -B "${BUILD_DIR}" -DCPP_STARTER_USE_FLTK=ON
-  cmake --build "${BUILD_DIR}" -j
 }
 
-function build_gtkmm() {
+function install_gtkmm() {
   apt-get install -y --no-install-recommends pkg-config libgtkmm-3.0-dev
-  BUILD_DIR="${1:-build-gtkmm}"
-  mkdir -p "${BUILD_DIR}"
-  cmake -S "${PROJECT_DIRNAME}" -B "${BUILD_DIR}" -DCPP_STARTER_USE_GTKMM=ON
-  cmake --build "${BUILD_DIR}" -j
 }
 
-function build_imgui() {
+function install_imgui() {
   apt-get install -y --no-install-recommends pkg-config libgl1-mesa-dev
-  BUILD_DIR="${1:-build-imgui}"
-  mkdir -p "${BUILD_DIR}"
-  cmake -S "${PROJECT_DIRNAME}" -B "${BUILD_DIR}" -DCPP_STARTER_USE_IMGUI=ON
-  cmake --build "${BUILD_DIR}" -j
 }
 
-function build_nana() {
+function install_nana() {
   # CMakeLists.txt is 'supposed' to install all of these automatically, but for
   # some reason it doesn't always work. Installing them manually does work.
   apt-get install -y --no-install-recommends libjpeg8-dev libpng-dev \
     libasound2-dev alsa-utils alsa-oss libx11-dev libxft-dev libxcursor-dev
-  BUILD_DIR="${1:-build-nana}"
-  mkdir -p "${BUILD_DIR}"
   # Note: Nana's headers trigger the -Wshadow warning, and cannot be compiled with -Werror.
   # Supposedly, this is fixed in the develop-1.8 branch, but I cannot confirm.
-  cmake -S "${PROJECT_DIRNAME}" -B "${BUILD_DIR}" -DCPP_STARTER_USE_NANA=ON -DWARNINGS_AS_ERRORS=FALSE
-  cmake --build "${BUILD_DIR}" -j
 }
 
-function build_qt() {
+function install_qt() {
   apt-get install -y --no-install-recommends qt5-default qtbase5-dev
```

Comment:
Qt is available from Conan. Not sure it is good practice to use the system-wide dependency! 
https://conan.io/center/qt

Reply:
I don't think it's a good idea to just use Conan for all dependency management; Conan is sometimes problematic, and I think it's best to give people options when they need them.

Depending on what parts of Qt you need, you could be downloading 17MB or multiple gigabytes. The Conan package doesn't appear to give you any options; it appears to install ~600MB, and takes a very long time to do it.

Not to plug my own projects, but there's also [aqtinstall](https://github.com/miurahr/aqtinstall) for installing Qt via command line (it works on Win/Mac/Linux), and [a Github Action](https://github.com/jurplel/install-qt-action) based on it. You can use either aqt or apt to install the minimum 17MB version, and IMHO if that's all you need that's a whole lot better than what Conan offers.

This change turns `build_examples.sh` into a documentation file, rather than a program that can be executed. If that's all it is now, I would rather have it tell you that you have multiple ways to install Qt, depending on your preferences.

Reply:
The benefit of using Conan is that the code will be cross-platform. Qt itself is cross-platform, and using apt for installing a cross-platform library is not the best way. This means that the developer should write separate build scripts for every possible platform. This is the exact problem that Conan fixes. 

Also, Conan has a caching system, so there are no worries regarding the repeated builds.